docs/guide/cookbook/module.md

```diff
@@ -419,7 +419,58 @@ Routes to be added must be _Array_ type even if it only has one element as you c
 
 ![route_liked_borderline](../images/route_liked.png)
 
-### 2-4. Recipe D (Use hooks)
+### 2-4. Recipe D (Create hooks)
+As you know we have two main kinds of hooks `listener` and `mutator`. In this section, we will look into how to use these hooks to create our own functionality.
+
+1. Create the `hooks.ts` inside `./src/modules/example-module`.
+
+2. Import `hooks` from the `@vue-storefront/core/lib/hooks`:
+
+```js
+import { createListenerHook, createMutatorHook } from '@vue-storefront/core/lib/hooks'
+```
+
+3. Prepare our own `hooks`:
+```ts
+// abridged ...
+
+const {
+  hook: beforeExampleHook,
+  executor: beforeExampleExecutor
+} = createMutatorHook()
+
+// Also we can add some generic types to hooks then the first object is responsible for incoming data structure and the second shows us how to return data will be looking.
+
+const {
+  hook: beforeExampleHook,
+  executor: beforeExampleExecutor
+} = createMutatorHook<{ clientItems: CartItem[], serverItems: CartItem[] }, any>()
+
+const {
+  hook: afterExampleHook,
+  executor: afterExampleExecutor
+} = createListenerHook()
+
+const exampleHooks = {
+  beforeExample: beforeExampleHook,
+  afterExample: afterExampleHook
+}
+const exampleHooksExecutors = {
+  beforeExample: beforeExampleExecutor,
+  afterExample: afterExampleExecutor
+}
+```
+
+4. Export created `hooks`:
+
+```js
+export {
+  exampleHooks,
+  exampleHooksExecutors
+}
+```
+
+### 2-5. Recipe E (Use hooks)
 One of the most intuitive way to build a module is using hooks. Open source creators more than often intentionally leave hooks as many as possible to everywhere they think extendable for 3rd party developers to inject logic into the flow of program. Here we will look into how _Vue Storefront_ did it its way.
```

Comment:
This intro is not proper anymore. I feel like 2-4 and 2-5 should be a one recipe. You should show how to create hooks, explain what's the executor,  why developer needs to call executor and how to call executor